Update DQC main page README (#114)
CDCgov/dibbs-query-connector: README.md

# DIBBs Query Connector

[![codecov](https://codecov.io/gh/CDCgov/phdi/branch/main/graph/badge.svg)](https://codecov.io/gh/CDCgov/phdi)
[![pre-commit](https://img.shields.io/badge/pre--commit-enabled-brightgreen?logo=pre-commit)](https://github.com/pre-commit/pre-commit)
[![Ruff](https://img.shields.io/endpoint?url=https://raw.githubusercontent.com/astral-sh/ruff/main/assets/badge/v2.json)](https://github.com/astral-sh/ruff)
[![Test github badge](https://github.com/CDCgov/phdi/actions/workflows/test.yaml/badge.svg)](https://github.com/CDCgov/phdi/actions/workflows/test.yaml)

**General disclaimer** This repository was created for use by CDC programs to collaborate on public health related projects in support of the [CDC mission](https://www.cdc.gov/about/organization/mission.htm). GitHub is not hosted by the CDC, but is a third party website used by CDC and its partners to share information and collaborate on software. CDC use of GitHub does not imply an endorsement of any one particular service, product, or enterprise.

## Overview

This repository is a part of the CDC/USDS [PHDI project](https://cdcgov.github.io/phdi-site/) and seeks to build the DIBBs Query Connector.

> The DIBBs Query Connector app offers a REST API and comprehensive UI for searching for a patient and viewing information tied to your case investigation and is fully customizable to meet the specific needs of public health investigators.

Reviewer: ```suggestion
The DIBBs Query Connector app offers a REST API and comprehensive UI for public health staff to query a wide network of healthcare organizations (HCOs), providing access to more complete and timely data. 

The Query Connector is a tool for public health practitioners to quickly retrieve patient records and relevant case information from HCOs. The tool works by connecting to Qualified Health Information Networks (QHINs) within the Trusted Exchange Framework and Common Agreement (TEFCA), ensuring immediate access to essential health data and facilitating timely public health decisions and interventions. Public health staff can interact with the Query Connector manually by entering simple patient details — such as name, date of birth, or medical identifiers — into a web-based search form. The tool also allows for automated queries via integration engines like Rhapsody and Mirth.  
```

### Problem Scope

> 
> Current public health systems that digest, analyze, and respond to data are siloed. Lacking access to actionable data, our national, as well as state, local, and territorial infrastructure, isn’t pandemic-ready. Our objective is to help the CDC best support PHAs in moving towards a modern public health data infrastructure. See our [public website](https://cdcgov.github.io/dibbs-site/) for more details.

Reviewer: ```suggestion

Current public health systems that digest, analyze, and respond to data are siloed. Lacking access to actionable data, our national, as well as state, local, and territorial infrastructure, isn’t pandemic-ready. Our objective is to help the CDC best support PHAs in moving towards a modern public health data infrastructure. See our [public website](https://cdcgov.github.io/dibbs-site/) for more details.

The current methods that public health jurisdictions use to collect data from HCOs can be time-consuming and cumbersome, often involving manual interventions like phone calls and faxes. The Query Connector aims to streamline the collection of health data using an intuitive querying process that can leverage QHINs within TEFCA or direct connections, thereby enabling data exchange from multiple HCOs for a variety of public health use cases such as newborn screening or STI case investigation. 
```

DIBBs Query Connector is a sibling project to

* [PHDI](https://github.com/CDCgov/phdi) to further help display, access, and interpret publich health data;
* [PRIME ReportStream](https://reportstream.cdc.gov), which focuses on improving the delivery of COVID-19 test data to public health departments; and
* [PRIME SimpleReport](https://simplereport.gov), which provides a better way for organizations and testing facilities to report COVID-19 rapid tests to public health departments.

## Implementation

**Implementation Support** - Resources to help users implement DIBBs Query Connector tools to manage their data.

> - **Examples** - Forthcoming!

Author: oh yeah, I forgot we had a postman for it. I'll take a look but yeah that wouldn't be surprising. Could be a good next backlog ticket to spruce it up

> - **Examples** - Forthcoming!

Author: to circle back, yeah, some of the sample calls no longer work 😕 , so we would probably want to do some additional cleanup / consideration if we want to move all those to be Hyper Unlucky.

created a follow-up ticket to track that aspect of work: https://linear.app/skylight-cdc/issue/QUE-52/update-postman-examples

> - **Examples** - Forthcoming!

Reviewer: Dang, thanks for making the new ticket!

- **Tutorials** - Forthcoming!

## Documentation

DIBBs Query Connector documentation, including instructions on how to install dependencies and run locally, is currently hosted within the repository, [here](https://github.com/CDCgov/dibbs-query-connector/blob/main/query-connector/README.md).

> Access to the production instance of the DIBBs Query Connector is available at [dibbs.cloud/tefca-viewer]().

Reviewer: ```suggestion
Access to the demo instance of the DIBBs Query Connector is available at [dibbs.cloud/tefca-viewer]().
```

## Additional Acknowledgments

We mapped the rootnames of the PHDI database to nicknames produced by the aggregation and synthesis of open source work from a number of projects. While we do not employ the packages and wrappers used by the various projects (merely their open source data), we wish to give credit to their various works building collections of nickname mappings. These projects are:

> 

Author: ahhh I had a sense some of that language didn't apply anymore but wanted to CYA just in case 😅 . removing!

> 

Author: removed!
